server/schedule/placement/fit_region_test.go

```diff
@@ -0,0 +1,329 @@
+			EndKey:   []byte(""),
+			Peers:    peers,
+		},
+		&metapb.Peer{Id: 1, StoreId: 1},
+	)
+	return region
+}
+
+func BenchmarkFitRegion(b *testing.B) {
+	i := 0
+	region := mockRegion(3, 0)
+	rules := []*Rule{
+		{
+			GroupID:        "pd",
+			ID:             "default",
+			Role:           Voter,
+			Count:          3,
+			LocationLabels: []string{},
+		},
+	}
+	storesSet := newMockStoresSet(100)
+	b.ResetTimer()
+	for i < b.N {
+		i++
+		FitRegion(storesSet, region, rules)
+	}
+}
+
+func BenchmarkFitRegionMoreStores(b *testing.B) {
+	i := 0
+	region := mockRegion(3, 0)
+	rules := []*Rule{
+		{
+			GroupID:        "pd",
+			ID:             "default",
+			Role:           Voter,
+			Count:          3,
+			LocationLabels: []string{},
+		},
+	}
+	storesSet := newMockStoresSet(200)
+	b.ResetTimer()
+	for i < b.N {
+		i++
+		FitRegion(storesSet, region, rules)
+	}
+}
+
+func BenchmarkFitRegionMorePeers(b *testing.B) {
+	i := 0
+	region := mockRegion(5, 0)
+	rules := []*Rule{
+		{
+			GroupID:        "pd",
+			ID:             "default",
+			Role:           Voter,
+			Count:          5,
+			LocationLabels: []string{},
+		},
+	}
+	storesSet := newMockStoresSet(100)
+	b.ResetTimer()
+	for i < b.N {
+		i++
+		FitRegion(storesSet, region, rules)
+	}
+}
+
+func BenchmarkFitRegionMorePeersEquals(b *testing.B) {
+	i := 0
+	region := mockRegion(3, 0)
+	rules := []*Rule{
+		{
+			GroupID:        "pd",
+			ID:             "default",
+			Role:           Leader,
+			Count:          1,
+			LocationLabels: []string{},
+		},
+		{
+			GroupID:        "pd",
+			ID:             "default-2",
+			Role:           Follower,
+			Count:          4,
+			LocationLabels: []string{},
+		},
+	}
+	storesSet := newMockStoresSet(100)
+	b.ResetTimer()
+	for i < b.N {
+		i++
+		FitRegion(storesSet, region, rules)
+	}
+}
+
+func BenchmarkFitRegionMorePeersSplitRules(b *testing.B) {
+	i := 0
+	region := mockRegion(3, 0)
+	rules := []*Rule{
+		{
+			GroupID:        "pd",
+			ID:             "default",
+			Role:           Leader,
+			Count:          1,
+			LocationLabels: []string{},
+		},
+		{
+			GroupID:        "pd",
+			ID:             "default-2",
+			Role:           Follower,
+			Count:          1,
+			LocationLabels: []string{},
+		},
+		{
+			GroupID:        "pd",
+			ID:             "default-3",
+			Role:           Follower,
+			Count:          1,
+			LocationLabels: []string{},
+		},
+		{
+			GroupID:        "pd",
+			ID:             "default-4",
+			Role:           Follower,
+			Count:          1,
+			LocationLabels: []string{},
+		},
+		{
+			GroupID:        "pd",
+			ID:             "default-5",
+			Role:           Follower,
+			Count:          1,
+			LocationLabels: []string{},
+		},
+	}
+	storesSet := newMockStoresSet(100)
+	b.ResetTimer()
+	for i < b.N {
+		i++
+		FitRegion(storesSet, region, rules)
+	}
+}
+
+func BenchmarkFitRegionMoreVotersSplitRules(b *testing.B) {
+	i := 0
+	region := mockRegion(5, 0)
+	rules := []*Rule{
+		{
+			GroupID:        "pd",
```

Comment:
Is it on purpose to use the same rule?

Reply:
yes, this is used to comparing the cpu resource consuming between less rules and more rules.

Reply:
Then how about using a variable?

Reply:
updated.